glidertest/tools.py

```diff
@@ -1117,16 +1111,19 @@ def ramsey_binavg(ds, var='VERT_CURR', zgrid=None, dz=None):
     ----
     I know this is a non-sensical name.  We should re-name, but is based on advice from Ramsey Harcourt.
     """
-
+    _necessary_variables_check(ds, [var, 'PRES'])
     press = ds.PRES.values
     ww = ds[var].values
 
     # Calculate depth from pressure using gsw
     if 'DEPTH_Z' in ds:
         depth = ds.DEPTH_Z.values
-    else:
+    elif 'DEPTH_Z' not in ds:

```

Comment:
`elif 'DEPTH_Z' not in ds` has no effect. By definition 'DEPTH_Z' is not in ds or the first if statement `if 'DEPTH_Z' in ds` would be True and this `elif` would never be reached. This should be `elif 'LATITUDE' in ds`

`elif` is tricky!